DSharpPlus/Entities/DiscordWebhookBuilder.cs

```diff
@@ -261,5 +261,29 @@ public void Clear()
             this._mentions.Clear();
             this._files.Clear();
         }
+
+        /// <summary>
+        /// Does the validation before we send a the Create/Modify request.
+        /// </summary>
+        /// <param name="isModify">Tells the method to perform the Modify Validation or Create Validation.</param>
+        internal void Validate(bool isModify = false)
+        {
+            if (isModify)
+            {
+                if (this.Files.Any())
+                    throw new ArgumentException("You cannot add files when modifying a message.");
+
+                if (this.Username.HasValue)
+                    throw new ArgumentException("You cannot change the username of a message.");
+
+                if (this.AvatarUrl.HasValue)
+                    throw new ArgumentException("You cannot change the avatar of a message.");
+            }
+            else
+            {
+                if (this.Files?.Count == 0 && string.IsNullOrEmpty(this.Content) && this.Embeds == null)
```

Comment:
`this.Embeds` will never be null, you should check for the collection length instead like you do for the files.